src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,29 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector, handleDelete) {
+    super({ popupSelector });
+    this._confirmElement = this._popupElement.querySelector(".form");
+    this._handleDelete = handleDelete;
+  }
+
+  _handleSubmit = () => {
+    this._handleDelete(this._data);
+  };
+
+  open = ({ deleter, _id }) => {
+    this._popupElement.classList.add("modal_opened");
+    this.setEventListeners();
```

Comment:
```js
    this._popupElement.classList.add("modal_opened");
    this.setEventListeners();
```

Please, delete this code

`super` is the reference to the parent class and you can call any method not to duplicate the code in your own (same) methods


```js

  open = ({ deleter, _id }) => {
    super.open();
```